tests/kafkatest/tests/core/security_test.py

```diff
@@ -87,16 +67,19 @@ def test_client_ssl_endpoint_validation_failure(self, security_protocol, interbr
         with hostname verification failure. Hence clients are expected to fail with LEADER_NOT_AVAILABLE.
         """
 
-        self.kafka.security_protocol = security_protocol
-        self.kafka.interbroker_security_protocol = interbroker_security_protocol
-        SecurityConfig.ssl_stores = TestSslStores(self.test_context.local_scratch_dir, valid_hostname=False)
+        SecurityConfig.ssl_stores = TestSslStores(self.test_context.local_scratch_dir,
+                                                  valid_hostname=False)
+
+        self.create_zookeeper()
+        self.zk.start()
 
+        self.create_kafka(security_protocol=security_protocol,
+                          interbroker_security_protocol=interbroker_security_protocol)
         self.kafka.start()
-        self.create_producer_and_consumer()
-        self.producer.log_level = "TRACE"
 
-        self.producer.start()
-        self.consumer.start()
+        # We need more verbose logging to catch the expected errors
+        self.create_and_start_clients(log_level="DEBUG")
```

Comment:
Not sure why it was TRACE before, do we only need DEBUG?

Reply:
Yeah, I reduced it because the message we were looking for was DEBUG level.